src/Controller/HearingController.php

```diff
@@ -462,4 +495,48 @@ public function resumeHearing(CaseEntity $case, Hearing $hearing): Response
 
         return $this->redirectToRoute('case_hearing_index', ['id' => $case->getId()]);
     }
+
+    /**
+     * @Route("/{case}/hearing/{hearingPost}/delete", name="case_hearing_post_delete")
+     */
+    public function hearingPostDelete(CaseEntity $case, HearingPost $hearingPost, DocumentUploader $documentUploader, CaseDocumentRelationRepository $relationRepository, Request $request): Response
+    {
+        $this->denyAccessUnlessGranted('edit', $case);
+
+        // Check that CSRF token is valid
+        if ($this->isCsrfTokenValid('delete'.$hearingPost->getId(), $request->request->get('_token'))) {
+            // Remove relation between case and document
+            $document = $hearingPost->getDocument();
+            $relation = $relationRepository->findOneBy(['case' => $case, 'document' => $document]);
+
+            if (null !== $relation) {
+                $this->entityManager->remove($relation);
+                $this->entityManager->flush();
+            }
+
+            // Remove file
+            $documentUploader->deleteDocumentFile($document);
+
+            // Set foreign keys between hearing post and document to null
```

Comment:
Consider rewording to “Remove references between post and document”, say. “Foreign keys” is something that the ORM handles and we don't really care about the techical details in our code.

Reply:
Reworded.